src/components/ProductForm/ProductForm.tsx

```diff
@@ -1,59 +1,64 @@
 import { LoaderOverlay } from 'components/LoaderOverlay/LoaderOverlay';
-import { ReactElement } from 'react';
+import { ReactElement, useMemo } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faPen } from '@fortawesome/free-solid-svg-icons';
-import { Dropdown } from 'components/Dropdown/Dropdown';
-import { useProductFormContext } from './ProductFormContext';
 import { Button } from 'components/Buttons/Button';
-import cn from 'classnames';
+import { useForm } from 'react-hook-form';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Select } from 'components/Inputs/Select';
+import { Input } from 'components/Inputs/Input';
+import { TextArea } from 'components/Inputs/TextArea';
+import { Checkbox } from 'components/Inputs/Checkbox';
+import { useProductFormContext } from './ProductFormContext';
+import { useCategories } from 'hooks/useCategories';
+import { useTags } from 'hooks/useTags';
+import { Label } from 'components/Inputs/Label';
+
+const formSchema = yup.object({
```

Comment:
are we able to refactor this into one utility and use it in the forms and add the messages as parameters

Reply:
not 100% sure how we could do this, each schema has values specific to the form and from what I understand we can't really create generic schemas, so we'd probably end up with multiple utils for the different forms. 

I guess to make this a bit cleaner I could move the schema and its type into a separate file e.g. `schemas/ProductFormSchema.ts` and import them into here?

Reply:
Ahh I see, yeah sure that sounds good!